service-wcs/src/main/java/org/oskari/wcs/request/DescribeCoverage.java

```diff
@@ -0,0 +1,58 @@
+package org.oskari.wcs.request;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.oskari.wcs.capabilities.Capabilities;
+import org.oskari.wcs.capabilities.Operation;
+import org.oskari.wcs.util.Maps;
+
+public class DescribeCoverage {
+
+    private DescribeCoverage() { /* Block */ }
+
+    /**
+     * Create query parameters for a GET encoded DescribeCoverage request You
+     * can use {@link #coverageExists(Capabilities, String)} to see if the
+     * coverage exists
+     *
+     * @param coverageId
+     *            id of the coverage
+     * @return Map with query parameters
+     */
+    public static Map<String, String> toQueryParameters(String coverageId) {
+        return Maps.of(
+                "service", "WCS",
+                "version", "2.0.1",
+                "request", "DescribeCoverage",
+                "coverageId", coverageId);
+    }
+
+    /**
+     * Get the HTTP endpoint declared in GetCapabilities for DescribeCoverage
+     * request
+     *
+     * @param wcs capabilities of the service
+     * @param get true if you want the GET endpoint, false if POST
+     * @return the endpoint which might not exist for your binding
+     */
+    public static Optional<String> getDescribeCoverageEndPoint(Capabilities wcs, boolean get) {
```

Comment:
Not really sure how I feel about the name "get". How would you feel about "useGet" or having two methods: one for get and another for post?